api/envoy/extensions/filters/listener/proxy_protocol/v3/proxy_protocol.proto

```diff
@@ -40,4 +40,9 @@ message ProxyProtocol {
 
   // The list of rules to apply to requests.
   repeated Rule rules = 1;
+
+  // NOTICE: only enable if ALL traffic to the listener comes from a trusted source.
+  // Defaults to false. If true, attempt to detect proxy protocol if present, and allow
+  // requests through if proxy protocol is not used on the connection.
+  bool detect_proxy_protocol = 2;
```

Comment:
I feel the major goal is allow the requests through if proxy protocol is not used. And actually the whole proxy protocol filter is about detect the proxy protocol, so this option name feels confuse. Should we call it something like `allow_requests_with_proxy_protocol`?

Reply:
+1 please clarify the name.